lib/ansible/plugins/callback/telegraf.py

```diff
@@ -0,0 +1,290 @@
+    CALLBACK_NAME = 'telegraf'
+    CALLBACK_NEEDS_WHITELIST = False
+
+    def __init__(self):
+        super(CallbackModule, self).__init__()
+        self._display.display('Playbook telegraf_callback init')
+
+        # the ansible controller running the callback
+        self._controller_hostname = socket.gethostname()
+
+
+        # Read config and set up API client
+        conf_path = os.path.join(os.path.dirname(__file__), 'telegraf_callback.yml')
+        params = self._load_conf(conf_path)
+        
+        # override config values with ENV vars if set
+        host = params.get('host', 'localhost')
+        if os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_HOST', None):
+            host = os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_HOST', None)
+        port = params.get('port', 8092)
+        if os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_PORT', None):
+            port = os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_PORT', None)
+        wire_protocol = params.get('wire_protocol', 'udp')
+        if os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_WIRE_PROTOCOL', None):
+            wire_protocol = os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_WIRE_PROTOCOL', None)
+        
+        self._metric_per_host = params.get('metric_per_host', True)
+        if os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_PER_HOST', None):
+            self._metric_per_host = os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_PER_HOST', None)
+        self._metric_prefix = params.get('metric_prefix', 'ansible')
+        if os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_PREFIX', None):
+            self._metric_prefix = os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_PREFIX', None)
+        self._metric_summary = params.get('metric_summary', False)
+        if os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_SUMMARY', None):
+            self._metric_summary = os.environ.get('ANSIBLE_TELEGRAF_CALLBACK_METRIC_SUMMARY', None)
+
+        if self._display.verbosity > 1:
+            print('Using wire protocol {}, host {}, port {}'.format(wire_protocol, host, port))
+        if wire_protocol.startswith('http'):
+            try:
+                from telegraf import HttpClient
+                self.client = HttpClient(host=host, port=port)
+            except ImportError:
+                print('WARNING: Could not import telegraf HttpClient please pip install pytelegraf[http].  Using udp')
+                self.client = TelegrafClient(host=host, port=port)                
+        else:
+            try:
+                from telegraf.client import TelegrafClient
+                self.client = TelegrafClient(host=host, port=port)
+            except ImportError as e:
+                print('Could not import telegraf.client please pip install pytelegraf')
+                raise e
+
+        self._playbook_name = None
+
+        self.start_time = datetime.now()        
+        self._options = None
```

Comment:
don't do this by default, set from cli `self._options` does not exist or is None